[fix/RF-35] Fix Icon

alex-radchenko-github opened the pull request:
[fix/RF-35] Fix Icon
Button.stories.tsx - add react import

alex-radchenko-github committed:
[fix/RF-35] Fix Icon
Button.stories.tsx - add react import

alex-radchenko-github committed:
[fix/RF-35] Fix Icon
Button.stories.tsx - add react import

MirgradR commented on src/shared/ui/Icon/Icon.tsx:
```diff
@@ -2,17 +2,19 @@ import React from 'react'
+      color={color}
+      preserveAspectRatio={preserveAspectRatio}
+      fill={fill}
+      {...(viewBox && { viewBox })}
```
мне кажется не нужно передавать, то что я тебе кидал самый оптимальный размер, а передача усложнит понимание компоненты


viewBox="0 0 24 24"
preserveAspectRatio="xMidYMid meet"

это не нужно передавать или хотябы сделай их дефолтными, но с возможностью передать

alex-radchenko-github replied:
Did it

MirgradR commented on src/shared/ui/Icon/Icon.model.ts:
```diff
@@ -1,8 +1,14 @@
 import icons from './assets'
 
+type IconType = keyof typeof icons
```
экспорт нужен, так как тип и размер могут понадобиться снаружи

alex-radchenko-github replied:
Did it

alex-radchenko-github committed:
[fix/RF-35] Fix Icon
Add exports type IconType and type IconSize
made default preserveAspectRatio and viewBox

alex-radchenko-github committed:
[fix/RF-35] Fix Icon
Add exports type IconType and type IconSize
made default preserveAspectRatio and viewBox

alex-radchenko-github commented:
Made changes

MirgradR committed:
[fix/RF-37] fix storybook config for svg loader

alex-radchenko-github committed:
[fix/RF-35] Fix Icon
rename icon.model.ts, icon.stories.ts

alex-radchenko-github committed:
[fix/RF-35] Fix Icon
rename icon.model.ts, icon.stories.ts

beckyuldashev committed:
[fix/RF-34] fix button component and stories

beckyuldashev committed:
Merge pull request #20 from MirgradR/fix/RF-37

[fix/RF-37] fix storybook config for svg loader

beckyuldashev committed:
Merge pull request #19 from MirgradR/fix/RF-34

[fix/RF-34] fix button component and stories

MirgradR committed:
fix
